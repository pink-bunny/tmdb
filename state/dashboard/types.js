export const FETCH_TRENDING_MOVIES = 'FETCH_TRENDING_MOVIES';
export const FETCH_TRENDING_MOVIES_SUCCESS = 'FETCH_TRENDING_MOVIES_SUCCESS';
export const FETCH_TRENDING_MOVIES_ERROR = 'FETCH_TRENDING_MOVIES_ERROR';
