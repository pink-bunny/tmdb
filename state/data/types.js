export const MERGE_MOVIES_LIST = 'MERGE_MOVIES_LIST';

export const MERGE_LISTS_LIST = 'MERGE_LISTS_LIST';
