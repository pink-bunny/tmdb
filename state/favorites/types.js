export const FETCH_FAVORITES = 'FETCH_FAVORITES';
export const FETCH_FAVORITES_SUCCESS = 'FETCH_FAVORITES_SUCCESS';
export const FETCH_FAVORITES_ERROR = 'FETCH_FAVORITES_ERROR';

export const TOGGLE_TO_FAVORITES = 'TOGGLE_TO_FAVORITES';
export const TOGGLE_TO_FAVORITES_SUCCESS = 'TOGGLE_TO_FAVORITES_SUCCESS';
export const TOGGLE_TO_FAVORITES_ERROR = 'TOGGLE_TO_FAVORITES_ERROR';
