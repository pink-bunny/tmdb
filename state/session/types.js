export const SESSION_REQUEST_TOKEN = 'SESSION_REQUEST_TOKEN';
export const SESSION_REQUEST_TOKEN_SUCCESS = 'SESSION_REQUEST_TOKEN_SUCCESS';
export const SESSION_REQUEST_TOKEN_ERROR = 'SESSION_REQUEST_TOKEN_ERROR';
