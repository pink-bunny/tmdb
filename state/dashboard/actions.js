import * as types from './types';

export const fetchTrendingMovies = (page = 1) => ({
  type: types.FETCH_TRENDING_MOVIES,
  page
});

export const fetchTrendingMoviesSuccess = (ids, totalItems, currentPage) => ({
  type: types.FETCH_TRENDING_MOVIES_SUCCESS,
  ids,
  totalItems,
  currentPage
});

export const fetchTrendingMoviesError = (error) => ({
  type: types.FETCH_TRENDING_MOVIES_ERROR,
  error
});

export const searchMovies = ({ search }, setErrors, setSubmitting, page = 1) => ({
  type: types.SEARCH_MOVIES,
  search,
  page,
  setErrors,
  setSubmitting
});
