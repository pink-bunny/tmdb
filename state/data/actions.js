import * as types from './types';
import { TOGGLE_TO_WATCHLIST_SUCCESS } from '../movie/types';

export const mergeMoviesList = (movies) => ({
  type: types.MERGE_MOVIES_LIST,
  movies
});

export const mergeListsList = (lists) => ({
  type: types.MERGE_LISTS_LIST,
  lists
});

export const toggleToWatchlistSuccess = (id, isMovieInWatchlist) => ({
  type: TOGGLE_TO_WATCHLIST_SUCCESS,
  id,
  isMovieInWatchlist
});
