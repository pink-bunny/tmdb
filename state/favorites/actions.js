import * as types from './types';

export const fetchFavorites = (page = 1) => ({
  type: types.FETCH_FAVORITES,
  page
});
export const fetchFavoritesSuccess = (ids, totalItems, currentPage) => ({
  type: types.FETCH_FAVORITES_SUCCESS,
  ids,
  totalItems,
  currentPage
});

export const fetchFavoritesError = (error) => ({
  type: types.FETCH_FAVORITES_ERROR,
  error
});

export const toggleToFavorite = (id, favorite, needRefetchList = false) => ({
  type: types.TOGGLE_TO_FAVORITES,
  id,
  favorite,
  needRefetchList
});

export const toggleToFavoriteSuccess = (id, isMovieInFavorites) => ({
  type: types.TOGGLE_TO_FAVORITES_SUCCESS,
  id,
  isMovieInFavorites
});
