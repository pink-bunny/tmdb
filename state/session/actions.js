import * as types from './types';

export const sessionRequestToken = () => ({
  type: types.SESSION_REQUEST_TOKEN
});

export const sessionRequestTokenSuccess = (token) => ({
  type: types.SESSION_REQUEST_TOKEN_SUCCESS,
  token
});
