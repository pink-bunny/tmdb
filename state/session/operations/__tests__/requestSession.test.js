import cookie from 'cookie_js';

import requestSessionLogic from '../requestSession';
import { REQUEST_SESSION } from '../../types';
import { mockMultiHttpClient } from '../../../../utils/testsHelpers/mockHttpClient';
import {
  authenticationTokenResponseSuccess,
  authenticationSessionResponseSuccess
} from '../__mocks__/requestSession';
import { requestSessionSuccess } from '../../actions';

jest.mock('cookie_js', () => ({
  set: jest.fn()
}));

jest.mock('../../actions', () => ({
  requestSessionSuccess: jest.fn()
}));

describe('requestSession logic', () => {
  const username = 'mock name';
  const password = '123456';
  const setErrors = jest.fn();
  const setSubmitting = jest.fn();
  const action = {
    type: REQUEST_SESSION,
    username,
    password,
    setErrors,
    setSubmitting
  };
  const dispatch = jest.fn();
  const done = jest.fn();
  let httpClient;
  httpClient = mockMultiHttpClient([
    { method: 'get', response: authenticationTokenResponseSuccess },
    { method: 'post', response: {} },
    { method: 'post', response: authenticationSessionResponseSuccess }
  ]);
  requestSessionLogic.process({ httpClient, action }, dispatch, done);

  it('matches snapshot', () => {
    expect(requestSessionLogic).toMatchSnapshot();
  });

  describe('httpClient', () => {
    const requestAuthTokenPath = 'authentication/token/new';
    const requestTokenPath = 'authentication/token/validate_with_login';
    const { data: responseToken } = authenticationTokenResponseSuccess;
    const requestToken = responseToken.request_token;
    const requestTokenBody = {
      username,
      password,
      request_token: requestToken /* eslint-disable-line */
    };
    const requestAuthSessionPath = 'authentication/session/new';
    const requestAuthSessionBody = {
      request_token: requestToken /* eslint-disable-line */
    };
    it('called with right arguments', () => {
      expect(httpClient.get).toHaveBeenNthCalledWith(1, requestAuthTokenPath);
      expect(httpClient.post).toHaveBeenNthCalledWith(1, requestTokenPath, requestTokenBody);
      expect(httpClient.post)
        .toHaveBeenNthCalledWith(2, requestAuthSessionPath, requestAuthSessionBody);
    });
  });

  describe('success.', () => {
    const { data: responseSessionId } = authenticationSessionResponseSuccess;
    const sessionId = responseSessionId.session_id;
    it('Dispatch requestSessionSuccess() with session id and username', () => {
      expect(requestSessionSuccess).toHaveBeenCalledWith(sessionId, username);
    });

    it('Set session id and username to cookie', () => {
      expect(cookie.set).toHaveBeenCalledWith({ sessionId, username });
    });
  });

  describe('failure.', () => {
    beforeAll(() => {
      httpClient = mockMultiHttpClient([
        { method: 'post', response: {}, reject: true }
      ]);
      requestSessionLogic.process({ httpClient, action }, dispatch, done);
    });

    it('setSubmitting called with false', () => {
      expect(setSubmitting).toHaveBeenCalledWith(false);
    });
    it('setErrors called with server error', () => {
      expect(setErrors).toHaveBeenCalledWith({ serverError: 'Username or password is wrong. Try again' });
    });
  });

  it('have been called done()', () => {
    expect(done).toHaveBeenCalled();
  });
});
