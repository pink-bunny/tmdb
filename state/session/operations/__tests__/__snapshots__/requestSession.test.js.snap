// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requestSession logic matches snapshot 1`] = `
Object {
  "cancelType": undefined,
  "debounce": 0,
  "latest": false,
  "name": undefined,
  "process": [Function],
  "processOptions": Object {
    "dispatchMultiple": true,
  },
  "throttle": 0,
  "transform": undefined,
  "type": "REQUEST_SESSION",
  "validate": undefined,
  "warnTimeout": 60000,
}
`;
